sdk/textanalytics/Azure.AI.TextAnalytics.Protocol/README.md

```diff
@@ -12,12 +12,307 @@ At this time, if you are looking to interact with the Azure Cognitive Services T
+
+```csharp
+var oneDoc = new JsonData(1);           // represents the document: 1
+int oneValue = (int)oneDoc;             // works, oneValue is 1
+
+var stringDoc = new JsonData("hello");  // represents the document: "hello"
+int stringValue = (string)stringDoc;     // works, stringValue is the string "hello"
+```
+
+If the `JsonData` represents an object, you can use the `Properties` property to list all the properties of an object:
+
+```csharp
+var objectDoc = new JsonData(new { key1 = "one", key2 = "two" }); // represents the document: { "key1": "one", "key2": "two" }
+Console.WriteLine(string.Join(',' objectDoc.Properties)); // prints "key1, key2"
+```
+
+And you can fetch a specific property by name, using the indexer:
+
+```csharp
+var objectDoc = new JsonData(new { key1 = "one", key2 = "two" });
+Console.WriteLine(objectDoc["key1"]); // prints "one"
+```
+
+Or by using `Get`, which will return "null" if the object does not have a property by that name (compared to the indexer which would throw an `InvalidOperationException` is the property was not present):
+
+```csharp
+var objectDoc = new JsonData(new { key1 = "one", key2 = "two" }); // represents the document: { "key1": "one", "key2": "two" }
+Console.WriteLine(objectDoc.Get("key1")));               // prints "one"
+Console.WriteLine(objectDoc.Get("missingKey") == null)); // prints "true"
+Console.WriteLine(objectDoc["missingKey"]);              // throws InvalidOperationException.
+```
+
+Arrays are similar to Objects, you can use the indexer to fetch a specific element from the array:
+
+```csharp
+var arrayDoc = new JsonData(new[] { "Hello", "JsonData" }); // represents the document: [ "Hello", "JsonData" ]
+Console.WriteLine(arrayDoc[0]);   // prints "Hello"
+Console.WriteLine(arrayDoc[1]);   // prints "JsonData"
+```
+
+You can also use the `Items` property to get an `IEnumerable<JsonData>`, which you can use to enumerate the values:
+
+```csharp
+var arrayDoc = new JsonData(new[] { "Hello", "JsonData" }); // represents the document: [ "Hello", "JsonData" ]
+foreach (JsonData item in arrayDoc.Items)
+{
+    Console.WriteLine(item);
+}
+```
+
+Combined, this provides a nice way to access values.  Consider the JsonData that represents the document:
+
+```json
+{
+  "students": [
+      {
+          "name": "Matt",
+          "address": [ "1 Microsoft Way", "Building 18", "Redmond, WA, 98034" ]
+      },
+      {
+          "name": "Bill",
+          "address": [ "1 Microsoft Way", "Building 34", "Redmond, WA, 98034" ]
+      }
+  ]
+}
+```
+
+You can access the inner values with ease:
+
+```csharp
+JsonData doc = JsonData.FromString(/* a string representing the above document */);
+Console.WriteLine(doc["students"][0]["name"]); // prints "Matt"
+Console.WriteLine(doc["students"][1]["address"][1]); // prints "Building 34"
+```
+
+##### Using `dynamic`
+
+You may also use the `dynamic` keyword with `JsonData`, and it behaves as you might expect:
```

Comment:
I would not say "as you might expect". I think many C# devs dont know dynamic. I think you should start this section by saying that dynamic allows foo["bar"] to be written as foo.bar.

Reply:
Fixed!